gprofiler/proc_events.py

```diff
@@ -96,8 +92,8 @@ def _proc_events_listener(self):
         """Runs forever and calls registered callbacks on process events"""
         try:
             self._socket.bind((os.getpid(), self._CN_IDX_PROC))
-        except socket.error:
-            logger.exception("")
+        except PermissionError as e:
+            raise PermissionError("You don't have permissions to bind to the process events connector") from e
```

Comment:
Don't need the extra layer IMO. Just let the original error pass

Reply:
The original error isn't very clear:
```
Traceback (most recent call last):
  File "<ipython-input-1-1a014f10004c>", line 98, in _proc_events_listener
    self._socket.bind((os.getpid(), self._CN_IDX_PROC))
PermissionError: [Errno 1] Operation not permitted
```
The description I added is nicer IMO.

Reply:
Both are permission errors and the stacktrace originates from `proc_events_listener`. Personally I dislike wrapping exceptions and prefer to view the original one, especially when it's not a-framework-inside-a-framework case (which might try to hide the original exception because it's too deep).

Anyway, since you `raise ... from e` I suppose it's fine, because the original info is kept. Up to you to decide

Reply:
> Anyway, since you raise ... from e I suppose it's fine

Yup, that's the idea